src/main/java/org/testng/internal/Parameters.java

```diff
@@ -52,11 +52,39 @@
       AfterMethod.class
   };
 
-  private static Map<String, List<Class>> mapping = Maps.newHashMap();
+  private static Map<String, List<Class<?>>> mapping = Maps.newHashMap();
+/*
+          +--------------+--------------+---------+--------+----------+-------------+
```

Comment:
@cbeust Its picking the `XmlTest` object that represents the last `<test>` tag found in the TestNG suite xml file and then showing its details. To be honest, the behavior is weird and TestNG should not allow a `XmlTest` injection into the `@BeforeSuite` annotation as you called out. But this will cause a backward compatibility issue if we fix this. Please let me know what needs to be done.